### PR TITLE
Use add() to append items to activeExports set in HarmonyExportImporedSpecifierDependency

### PR DESCRIPTION
**Pull request by mohsen1**
**What kind of change does this PR introduce?**
Bug fix

**Did you add tests for your changes?**
no

**If relevant, link to documentation update:**
n/a

**Summary**
This is part of https://github.com/webpack/webpack/pull/6862 

TypeScript compiler found that we're calling `concat` on a `Set` here

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

**Comment by ooflorent** [comment]
What about the `length` test just above it?

File: lib/dependencies/HarmonyExportImportedSpecifierDependency.js [hunk 1]
```diff
@@ -567,10 +567,10 @@ HarmonyExportImportedSpecifierDependency.Template = class HarmonyExportImportedS
 
 				// Filter out exports which are defined by other exports
 				// and filter out default export because it cannot be reexported with *
-				if (activeExports.length > 0)
+				if (activeExports.size > 0)
 					content +=
 						"if(" +
-						JSON.stringify(activeExports.concat("default")) +
+						JSON.stringify(activeExports.add("default")) +
```
Comment (ooflorent) [inline lib/dependencies/HarmonyExportImportedSpecifierDependency.js]: `activeExports ` must be serialized into an array:

```js
Array.from(activeExports).concat("default")
```

File: lib/dependencies/HarmonyExportImportedSpecifierDependency.js [hunk 2]
```diff
@@ -567,10 +567,10 @@ HarmonyExportImportedSpecifierDependency.Template = class HarmonyExportImportedS
 
 				// Filter out exports which are defined by other exports
 				// and filter out default export because it cannot be reexported with *
-				if (activeExports.length > 0)
+				if (activeExports.size > 0)
 					content +=
 						"if(" +
-						JSON.stringify(activeExports.concat("default")) +
+						JSON.stringify(Array.from(activeExports.add("default"))) +
```
Comment (sokra) [inline lib/dependencies/HarmonyExportImportedSpecifierDependency.js]: Technically this is correct, but using the return value of `add` is kind of weird to me.

Better use `JSON.stringify(Array.from(activeExports).concat("default"))`

**Comment by ooflorent** [comment]
@mohsen1 it looks like you didn’t fix what Tobias asked for. Was there a merged conflict wrongly resolved?

**Comment by sokra** [comment]
Only integration test contribute to coverage. Unittests are just for fun.

Code like this could trigger that change:

``` js
export var x = 1;
export * from ",/cjs";
```

``` js
// cjs.js
module.exports = { x: 2, y: 3 };
```

File: test/cases/parsing/harmony-export-import-specifier/index.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+"use strict";
+
+import {x, y} from "./a";
+
+it("namespace export as from commonjs should override named export", function() {
```
Comment (mohsen1) [inline test/cases/parsing/harmony-export-import-specifier/index.js]: @sokra @domenic
Actually I'm not sure what the spec is here. Should namespace export override the named export?

Comment (sokra) [reply test/cases/parsing/harmony-export-import-specifier/index.js]: No, the spec says `*` export should have lower precedence than direct export. Order of exports doesn't matter here.

So the correct result would be x = 1

**Comment by sokra** [comment]
I fixed it for you

**Comment by sokra** [comment]
Thanks

**Pull request merged**